### PR TITLE
feat: make appointment  type component

### PR DESCRIPTION
**Pull request by Lilly-nn**
## Describe your changes

- I made an appointment first page and added component where the user can select type of appointments he/she needs. Added 'cancel appointment' modal.

image
image
If the user doesn't fill out all fields, submit button is inactive:
image
image
image
image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-76)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/create-appointment/AppointmentType.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  AppointmentTypeText,
+  Background,
+  IconWrapper,
+} from './styles';
+import { AppointmentTypeI } from './types';
+
+enum Appointment {
+  oneTime = 'one-time',
+  recurring = 'recurring',
+}
```
Comment (Simpsonkk) [inline src/components/create-appointment/AppointmentType.tsx:21]: It's better to put enum in a separate file as we do with constants

File: src/components/create-appointment/AppointmentType.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          size="small"
+          value={appointmentName}
+          onChange={(e): void => setAppointmentName(e.target.value)}
+        />
+        {appointmentName.length > MAX_APPOINTMENT_NAME_LENGTH && (
+          <ErrorText>{translate('create_appointment.errors.max_type_char')}</ErrorText>
+        )}
+        <AppointmentTypeCard
+          className={appointmentType === Appointment.oneTime ? 'active' : ''}
+          onClick={(): void => setAppointmentType(Appointment.oneTime)}
```
Comment (Simpsonkk) [inline src/components/create-appointment/AppointmentType.tsx:45]: It's better not to use inline functions. It makes the code easier to read and maintain.

File: src/components/book-appointment/styles.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const IconBackground = styled('div')`
+  background: ${PRIMARY.white};
+  padding: 44px 48px;
+  border-radius: 50%;
+`;
+
+const MainText = styled(Typography)`
+  color: ${PRIMARY.black};
+  font-size: ${TYPOGRAPHY.base}px;
+  font-weight: 500;
```
Comment (ProggerOleg) [inline src/components/book-appointment/styles.ts:31]: font-weight should be taken from typography

File: src/components/book-appointment/styles.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  gap: 8px;
+  align-items: center;
+`;
+
+const Button = styled(FilledButton)`
+  width: 82%;
+  padding: 8px;
+  font-size: ${TYPOGRAPHY.base_xs}px;
+  text-transform: none;
+  font-weight: 500;
```
Comment (ProggerOleg) [inline src/components/book-appointment/styles.ts:52]: Same here

**Review by lyudad — APPROVED** [review]

**Pull request merged**